specs/_features/eip7594/p2p-interface.md

```diff
@@ -0,0 +1,180 @@
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Modifications in EIP-7594](#modifications-in-eip-7594)
+  - [Preset](#preset)
+  - [Configuration](#configuration)
+  - [Containers](#containers)
+    - [`DataColumnSidecar`](#datacolumnsidecar)
+    - [`DataColumnIdentifier`](#datacolumnidentifier)
+  - [Helpers](#helpers)
+      - [`verify_data_column_sidecar_kzg_proof`](#verify_data_column_sidecar_kzg_proof)
+      - [`verify_data_column_sidecar_inclusion_proof`](#verify_data_column_sidecar_inclusion_proof)
+      - [`compute_subnet_for_data_column_sidecar`](#compute_subnet_for_data_column_sidecar)
+  - [The gossip domain: gossipsub](#the-gossip-domain-gossipsub)
+    - [Topics and messages](#topics-and-messages)
+      - [Samples subnets](#samples-subnets)
+        - [`data_column_sidecar_{subnet_id}`](#data_column_sidecar_subnet_id)
+  - [The Req/Resp domain](#the-reqresp-domain)
+    - [Messages](#messages)
+      - [DataColumnSidecarByRoot v1](#datacolumnsidecarbyroot-v1)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Modifications in EIP-7594
+
+### Preset
+
+| Name                                     | Value                             | Description                                                         |
+|------------------------------------------|-----------------------------------|---------------------------------------------------------------------|
+| `KZG_COMMITMENTS_INCLUSION_PROOF_DEPTH`   | `uint64(floorlog2(get_generalized_index(BeaconBlockBody, 'blob_kzg_commitments')))` (= 4) | <!-- predefined --> Merkle proof index for `blob_kzg_commitments` |
+
+### Configuration
+
+| Name                                     | Value                             | Description                                                         |
+|------------------------------------------|-----------------------------------|---------------------------------------------------------------------|
+| `DATA_COLUMN_SIDECAR_SUBNET_COUNT`              | `32`                               | The number of data column sidecar subnets used in the gossipsub protocol.  |
+
+### Containers
+
+#### `DataColumnSidecar`
+
+```python
+class DataColumnSidecar(Container):
+    index: ColumnIndex  # Index of column in extended matrix
+    column: DataColumn
+    kzg_commitments: List[KZGCommitment, MAX_BLOB_COMMITMENTS_PER_BLOCK]
+    kzg_proofs: List[KZGProof, MAX_BLOB_COMMITMENTS_PER_BLOCK]
+    signed_block_header: SignedBeaconBlockHeader
+    kzg_commitments_inclusion_proof: Vector[Bytes32, KZG_COMMITMENTS_INCLUSION_PROOF_DEPTH]
+```
+
+#### `DataColumnIdentifier`
+
+```python
+class DataColumnIdentifier(Container):
+    block_root: Root
+    index: ColumnIndex
+```
+
+### Helpers
+
+##### `verify_data_column_sidecar_kzg_proof`
+
+```python
+def verify_data_column_sidecar_kzg_proof(sidecar: DataColumnSidecar) -> bool:
```

Comment:
I think we should rename this to `verify_data_column_sidecar_kzg_proofs`, with plural proofs.

Reply:
I don't have a strong opinion on this, but from another point of view, it can be seen as a proof of the sidecar rather than proofs of cells.